File: wpaxos/test/test_cuckoo_hash_table.cpp
#include "../src/cuckoo_hash_table.h"
#include <cassert>

void test_insert_and_find() {
    CuckooHashTable hashTable;
    hashTable.insert(42);
    assert(hashTable.find(42));
}

int main() {
    test_insert_and_find();
    return 0;
}

File: epaxos/test/test_cuckoo_hash_table.rs
#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_insert_and_find() {
        let mut table = CuckooHashTable::new();
        
        table.insert(5);
        table.insert(10);

        assert!(table.find(5));
        assert!(table.find(10));
        assert!(!table.find(15));
    }

}